sklearn/utils/tests/test_extmath.py

```diff
@@ -330,6 +331,36 @@ def test_randomized_svd_sign_flip():
         assert_almost_equal(np.dot(v2.T, v2), np.eye(2))
 
 
+def test_randomized_svd_sign_flip_with_transpose():
+    # Check if flipping is always done based on u
+    # irrespective of transpose.
+    # See https://github.com/scikit-learn/scikit-learn/issues/5608
+    # for more details.
+    def max_loading_is_positive(u, v):
+        # returns bool tuple indicating if
+        # the values maximising np.abs
+        # are positive across all rows
+        # for u and across all columns for v.
+        return((np.abs(u).max(axis=0) == u.max(axis=0)).all(),
```

Comment:
Just for readability I would do something like:

```python

u_based = (np.abs(u).max(axis=0) == u.max(axis=0)).all()
v_based = (np.abs(v).max(axis=1) == v.max(axis=1)).all()

return u_based, v_based
```

You could also put your comment as the max_loading_is_positive function docstring.